hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1547,3 +1547,66 @@ def hpat_pandas_series_nunique_impl(self, dropna=True):
             return len(data_set) + 1
 
     return hpat_pandas_series_nunique_impl
+
+
+@overload_method(SeriesType, 'corr')
+def hpat_pandas_series_corr(self, other, method='pearson', min_periods=None):
+    """
+    Pandas Series method :meth:`pandas.Series.corr` implementation.
+
+    Note: Unsupported mixed numeric and string data
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_corr
+
+    Parameters
+    -----------
+    self: :obj:`pandas.Series`
+        input series
+    other: :obj:`pandas.Series`
+        input series
+    method:
+        *unsupported
+    min_periods: :obj:`int`, default None
+
+    Returns
+    -------
+    :obj:`float`
+         returns :obj:`float` object
+    """
+
+    _func_name = 'Method corr().'
+
+    if not (isinstance(self, SeriesType) and isinstance(other, SeriesType)):
+        raise TypingError('{} The object must be a pandas.series. Given: {} and {}'.format(_func_name, self, other))
+
+    if not (isinstance(self.data.dtype, types.Number) and isinstance(other.data.dtype, types.Number)):
+        raise TypingError(
+            'Currently function supports only numeric values. Given: {} and {}'.format(_func_name, self, other))
+
+    if not isinstance(min_periods, (types.Number, types.Omitted, types.NoneType)):
```

Comment:
I would say type of the parameter should be checked on `integer` instead of `number`.